blog/2020-10-23-the-making-of-new-user-sdk-tutorials.md

```diff
@@ -0,0 +1,110 @@
+2. In the same way that the code must be optimized for the docs, the docs must also be optimized for the code. In essence, the docs are being “driven” and "defined" by the underlying code. And if no one has coined the term yet, I think the credit for “code-driven documentation” should go to our Head of Product, Ryland Goldstein as he pinged me early one morning to share that epiphany with me.
+
+We decided to embrace [Snipsync](https://github.com/temporalio/snipsync) and solve the challenges that it introduced.
+
+## Template repos
+
+<img alt="image" class="docs-image-centered docs-image-max-width-20" src="https://raw.githubusercontent.com/temporalio/documentation-images/main/static/templates.png" />
+
+We now had a way to synchronize code with our documentation. But from where will the code be synchronized? We know users will likely want to view the source file and relative file path of the code snippet for added context. They will also be likely to clone the repo and try to run the sample.
+
+We actually already repositories of code samples for the [Go SDK](https://github.com/temporalio/samples-go) and [Java SDK](https://github.com/temporalio/samples-java). And while we desired to see more samples, there were already quite a few of them in each. But, we discovered that shared sample repositories tended to have two issues that made them less ideal for synchronizing with docs.
+
+1. While convenient, colocating multiple samples in a single repo is far less approachable compared with storing samples in self-contained repos.
+2. In shared sample repositories, it is harder to retain the idiomatic nature of any language and mirror the functionality of any sample across different languages.
+
+So for each of the sample applications we planned on using to drive our documentation, we created a corresponding template repo. These template repos can be easily copied, built, and run in a matter of minutes.
+
+- [Go money transfer template repo](https://github.com/temporalio/money-transfer-project-template-go)
+- [Go hello world template repo](https://github.com/temporalio/hello-world-project-template-go)
+- [Java money transfer template repo](https://github.com/temporalio/money-transfer-project-template-java)
+- [Java hello world template repo](https://github.com/temporalio/hello-world-project-template-java)
+
+## The tutorial
+
+<img alt="image" class="docs-image-centered docs-image-max-width-20" src="https://raw.githubusercontent.com/temporalio/documentation-images/main/static/tutorials.png" />
+
+Since the goal of our documentation changes was to help with new user acquisition, we decided to aim for a "tutorial" style of documentation. The first iterations aimed to build upon and replace the existing SDK “quick start” pages that maintained the status quo and printed “Hello World!” to the console. As you may have guessed, this route wasn’t sufficient enough to show users the real values that Temporal offers.
```

Comment:
```suggestion
Since the goal of our documentation changes was to help with new user acquisition, we decided to aim for a "tutorial" style of documentation. The first iterations aimed to build upon and replace the existing SDK “quick start” pages that maintained the status quo and printed “Hello World!” to the console. As you may have guessed, this route wasn’t sufficient enough to show users the real value Temporal offers.
```